base.rkt

```diff
@@ -73,25 +77,191 @@
   (if pos (- len (length pos)) #f))
 
 ; (-> (or/c (is-a?/c bitmap%) pict?) pict?)
+; creates a dark-gray/light-gray grid to place behind images
+; so that if they are transparent, the grid will become visible.
+#|(define (transparency-grid img)
+  (define dgray-color (make-object color% 128 128 128))
+  (define lgray-color (make-object color% 204 204 204))
+  (define x (if (pict? img) (pict-width img) (send img get-width)))
+  (define y (if (pict? img) (pict-height img) (send img get-height)))
+  ; 10 for a square's width
+  (define x-times (inexact->exact (floor (/ x 10))))
+  ; 20 for both squares' height
+  (define y-times (inexact->exact (floor (/ y 20))))
+  (define dgray-square (filled-rectangle 10 10 #:color dgray-color #:draw-border? #f))
+  (define lgray-square (filled-rectangle 10 10 #:color lgray-color #:draw-border? #f))
+  (define aline
+    (apply hc-append
+           (for/list ([times (in-range x-times)]
+                      [i (in-naturals)])
+             (if (even? i)
+                 dgray-square
+                 lgray-square))))
+  (define bline
+    (apply hc-append
+           (for/list ([times (in-range x-times)]
+                      [i (in-naturals)])
+             (if (even? i)
+                 lgray-square
+                 dgray-square))))
+  (apply vl-append (make-list y-times (vl-append aline bline))))|#
+
+; (-> (or/c (is-a?/c bitmap%) pict?) pict?)
+; creates a dark-gray/light-gray grid to place behind images
+; so that if they are transparent, the grid will become visible.
 (define (transparency-grid img)
   (define dgray-color (make-object color% 128 128 128))
   (define lgray-color (make-object color% 204 204 204))
   (define x (if (pict? img) (pict-width img) (send img get-width)))
   (define y (if (pict? img) (pict-height img) (send img get-height)))
-  (define x-times (inexact->exact (floor (/ x 20))))
+  ; 10 for a square's width
+  (define x-times (inexact->exact (floor (/ x 10))))
+  ; 20 for both square's height
   (define y-times (inexact->exact (floor (/ y 20))))
+  (define remainder-x (modulo (floor x) 10))
+  (define remainder-y (modulo (floor y) 20))
   (define dgray-square (filled-rectangle 10 10 #:color dgray-color #:draw-border? #f))
   (define lgray-square (filled-rectangle 10 10 #:color lgray-color #:draw-border? #f))
-  (define vboth (vl-append dgray-square lgray-square))
-  (define aline (apply hc-append (make-list x-times (hc-append dgray-square lgray-square))))
-  (define bline (apply hc-append (make-list x-times (hc-append lgray-square dgray-square))))
-  (apply vl-append (make-list y-times (vl-append aline bline))))
+  ; remainder square for width
+  (define rdsquare-x
```

Comment:
I'm having trouble following all this remainder size logic below. I'll have to review this in more detail.